lib/ex_aws/s3/utils.ex

```diff
@@ -84,6 +84,41 @@ defmodule ExAws.S3.Utils do
     {permission, grants}
   end
 
+  def build_notification_config(sections) do
+    properties = Enum.map(sections, &build_notification_config_section/1)
+    ["<NotificationConfiguration>", properties, "</NotificationConfiguration>"]
+    |> IO.iodata_to_binary
+  end
+
+  def build_notification_config_section(config) do
+    types = [
+      topic: "TopicConfiguration",
+      queue: "QueueConfiguration",
+      cloud_function: "CloudFunctionConfiguration"
+    ]
+    arn = [
+      topic: "Topic",
+      queue: "Queue",
+      cloud_function: "CloudFunction"
+    ]
+    ["<#{types[config.type]}>",
+     "<Id>", config.id, "</Id>",
+     build_notification_config_filter(config[:filter]),
+     "<#{arn[config.type]}>", config.arn, "</#{arn[config.type]}>",
+     Enum.map(config.events, &("<Event>#{&1}</Event>")),
+     "</#{types[config.type]}>"]
+  end
```

Comment:
Hey there! First off <3 for working on this, it's greatly appreciated.
I'd like to do this mapping of types / arn a little differently. I think we either should simply camelize whatever they give us and let AWS validate that it's correct, or we ought to enforce that they give a value for which we have a key.

Right now if they give `topic_configuration` for example because that's what they think it should be then the value that gets sent to AWS will be `nil`, and they won't know why easily.
nil, which will be confusing.